pkg/models/type.go

```diff
@@ -0,0 +1,16 @@
+package models
+
+type Type struct {
+	Personal bool
+	Team     bool
+}
```

Comment:
I think that `Type` shouldn't to be a structure with two fields. It would be a lot simpler being a type to a string or a value, and create possible values from this type.

For example:


```go
type Type string

const TypePersonal Type = "personal"

const TypeTeam Type = "team"
```

Reply:
In our Namespace structure, we consistently use pointer indirection for 
foreign types/structures. However, `'Type'` field was a raw string, breaking 
this pattern.

``` go

type Namespace struct {
	Name         string             `json:"name"  validate:"required,hostname_rfc1123,excludes=.,lowercase"`
	Owner        string             `json:"owner"`
	TenantID     string             `json:"tenant_id" bson:"tenant_id,omitempty"`
	Members      []Member           `json:"members" bson:"members"`
	Settings     *NamespaceSettings `json:"settings"`
	Devices      int                `json:"-" bson:"devices,omitempty"`
	Sessions     int                `json:"-" bson:"sessions,omitempty"`
	MaxDevices   int                `json:"max_devices" bson:"max_devices"`
	DevicesCount int                `json:"devices_count" bson:"devices_count,omitempty"`
	CreatedAt    time.Time          `json:"created_at" bson:"created_at"`
	Billing      *Billing           `json:"billing" bson:"billing,omitempty"`
	Type         *Type              `json:"type" bson:"type,omitempty"`
}
```

By wrapping `'Type'` in a struct:
- Maintains consistent pointer indirection across codebase
- Makes architectural pattern explicit and uniform 
- Simplifies future refactoring by following same memory model
- Reduces cognitive load when working with the codebase

should `Type` be a struct of a single string, for keeping this pattern, what you think?

Reply:
With the suggestion I gave, the Type wouldn't be a raw string, but a type "over" a string, what means that a variable/parameter from this "Type" would only accept values from this type, how I have constructed in the example.

About the points you make about the structure, this is my view:

> Maintains consistent pointer indirection across codebase

The "Type" only stores a value, so a structure doesn't make sense to exist for this value.

> Makes architectural pattern explicit and uniform

It isn't an architecture pattern enforced, it just appears from the necessities.

> Simplifies future refactoring by following same memory model

It cannot simplify something that doesn't exist, or maybe will never exist yet.

> Reduces cognitive load when working with the codebase

Yeah, a simple "string" could reduce the cognitive load of a structure with one value.


Check the `/pkg/models/member.go` to see an existent example.